connector/src/main/java/org/jboss/as/connector/util/ModelNodeUtil.java

```diff
@@ -41,16 +65,36 @@ public static String getResolvedStringIfSetOrGetDefault(final OperationContext c
 
     }
 
-    public static Extension extractExtension(final OperationContext operationContext, final ModelNode dataSourceNode,
-                                             final SimpleAttributeDefinition className, final PropertiesAttributeDefinition propertyName)
+    public static Extension extractExtension(final OperationContext operationContext, final ModelNode dataSourceNode, final SimpleAttributeDefinition classNameAttribute,
+                                             final PropertiesAttributeDefinition propertyNameAttribute) throws ValidateException, OperationFailedException {
+        return extractExtension(operationContext, dataSourceNode, classNameAttribute, null, propertyNameAttribute);
+    }
+
+    public static Extension extractExtension(final OperationContext operationContext, final ModelNode dataSourceNode, final SimpleAttributeDefinition classNameAttribute,
+                                             final SimpleAttributeDefinition moduleNameAttribute, final PropertiesAttributeDefinition propertyNameAttribute)
             throws ValidateException, OperationFailedException {
-        if (dataSourceNode.hasDefined(className.getName())) {
-            String exceptionSorterClassName = getResolvedStringIfSetOrGetDefault(operationContext, dataSourceNode, className);
+        if (dataSourceNode.hasDefined(classNameAttribute.getName())) {
+            String className = getResolvedStringIfSetOrGetDefault(operationContext, dataSourceNode, classNameAttribute);
+
+            String moduleName = null;
+            if (moduleNameAttribute != null) {
+                moduleName = getResolvedStringIfSetOrGetDefault(operationContext, dataSourceNode, moduleNameAttribute);
+            }
+
+            ClassLoader moduleClassLoader = null;
+            if (moduleName != null) {
+                try {
+                    final ModuleIdentifier exceptionSorterModuleId = ModuleIdentifier.create(moduleName);
+                    moduleClassLoader = Module.getCallerModuleLoader().loadModule(exceptionSorterModuleId).getClassLoader();
+                } catch (ModuleLoadException exception) {
+                    throw new OperationFailedException("Wrong module name", exception);
```

Comment:
@tadamski This needs to be an i18n message, and should include the faulty module name.

Reply:
fixed